### PR TITLE
vmss: (preview) support rolling upgrade

### PR DESCRIPTION
**Pull request by yugangw-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -800,6 +800,12 @@ def list_features(client, resource_provider_namespace=None):
     return client.list_all()
 
 
+def register_feature(client, resource_provider_namespace, feature_name):
+    logger.warning("Once the fearure '{}' is registered, invoking 'az provider register -n {}' is required "
+                   "to get the change propagated".format(resource_provider_namespace, feature_name))
+    return client.register(resource_provider_namespace, feature_name)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: I am in the conversation with ARM team for whether it is a service bug. If yes, I will revert the change

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py [hunk 2]
```diff
@@ -314,7 +314,7 @@ def test_custom_image(self, resource_group):
         ])
         self.cmd('vm show -g {} -n vm1'.format(resource_group), checks=[
             JMESPathCheckV2('storageProfile.imageReference.resourceGroup', resource_group),
-            JMESPathCheckV2('storageProfile.osDisk.createOption', 'fromImage')
+            JMESPathCheckV2('storageProfile.osDisk.createOption', 'FromImage')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py]: This is a recent swagger change brought in from the new azure-mgmt-compute

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py [hunk 3]
```diff
@@ -1763,10 +1762,10 @@ def body(self):
                  ' --vnet-name {} --subnet {} -l "West US" --vm-sku {}'
                  ' --storage-container-name {} -g {} --name {} --load-balancer {}'
                  ' --ssh-key-value \'{}\' --backend-pool-name {}'
-                 ' --nat-pool-name {} --use-unmanaged-disk'
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/test_vm_commands.py]: This worked before by coincidence, as it was just ignored in the temple builder

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 4]
```diff
@@ -800,6 +800,12 @@ def list_features(client, resource_provider_namespace=None):
     return client.list_all()
 
 
+def register_feature(client, resource_provider_namespace, feature_name):
+    logger.warning("Once the fearure '{}' is registered, invoking 'az provider register -n {}' is required "
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: Typo: fearure

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py [hunk 5]
```diff
@@ -274,6 +274,11 @@
     short-summary: Place the CLI in a waiting state until a condition of a scale set is met.
 """
 
+helps['vmss rolling-ugrade'] = """
+    type: group
+    short-summary: (Preview) Manage rolling upgrades.
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py]: Elsewhere, we have Preview in all caps.
Should we be consistent throughout?

**Comment by yugangw-msft** [comment]
FYI, below are the new commands, which are all auto-command based. The main works is on `vmss create`
```bash
(env) C:\SDK\azure-cli>az vmss rolling-upgrade -h

Group
    az vmss rolling-upgrade

Commands:
    cancel          : Cancels the current virtual machine scale set rolling upgrade.
    get-latest      : Gets the status of the latest virtual machine scale set rolling upgrade.
    start-os-upgrade: Starts a rolling upgrade to move all virtual machine scale set instances to
                      the latest available Platform Image OS version.
```

**Pull request merged**